### PR TITLE
4844: read blobs as raw bytes not as unit256 and fix blob versioned hashes validation

### PR DESCRIPTION
**Pull request by fab-10**
Signed-off-by: Fabio Di Fabio <fabio.difabio@consensys.net>

## Documentation

- [ ] I thought about documentation and added the `doc-change-required` label to this PR if
    [updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).

## Acceptance Tests (Non Mainnet)

- [ ] I have considered running `./gradlew acceptanceTestNonMainnet` locally if my PR affects non-mainnet modules.

## Changelog

- [ ] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

File: evm/src/main/java/org/hyperledger/besu/evm/precompile/KZGPointEvalPrecompiledContract.java [hunk 1]
```diff
@@ -49,53 +52,55 @@ public KZGPointEvalPrecompiledContract() {
...
+      if (pathToTrustedSetup.isPresent()) {
+        Path pathToSetup = pathToTrustedSetup.get();
+        absolutePathToSetup = pathToSetup.toAbsolutePath().toString();
       } else {
-        throw new IllegalArgumentException("provided file not a setup for either 4 or 4096 bits");
+        InputStream is =
+            KZGPointEvalPrecompiledContract.class.getResourceAsStream(
+                "mainnet_kzg_trusted_setup_4096.txt");
+        try {
+          File jniWillLoadFrom = File.createTempFile("kzgTrustedSetup", "txt");
```
Comment (jflo) [inline evm/src/main/java/org/hyperledger/besu/evm/precompile/KZGPointEvalPrecompiledContract.java:66]:
```suggestion
          File jniWillLoadFrom = File.createTempFile("kzgTrustedSetup", ".txt");
```

**Review by jflo — APPROVED** [review]
minor suggestion, looks good otherwise.

**Comment by jflo** [comment]
Unit tests are failing and it is due to a coredump happening when the new test tries to load the new trusted setup file. If I try to run that test with a known good setup file, it does not coredump.

**Review by jflo — CHANGES_REQUESTED** [review]
Unit tests are failing and it is due to a coredump happening when the new test tries to load the new trusted setup file. If I try to run that test with a known good setup file, it does not coredump.

**Review by gezero — APPROVED** [review]

**Comment by fab-10** [comment]
> Unit tests are failing and it is due to a coredump happening when the new test tries to load the new trusted setup file. If I try to run that test with a known good setup file, it does not coredump.

I did some debug and found that the c-kzg lib does not support mixing `MINIMAL` and `MAINNET` preset in the same process, since each one of them as a standalone native lib, so the test loads first the minimal lib, and when we try to load a mainnet trusted setup used by `aggregateKzgProof` it fails.
I do not know if there is a workaround for this, so for now I am putting ignore on this test

**Pull request merged**